compiler/src/iree/compiler/Codegen/Dialect/GPU/Transforms/Transforms.cpp

```diff
@@ -923,13 +923,32 @@ struct LowerShuffleTensor
                                 PatternRewriter &rewriter) const final {
     Location loc = shuffleOp.getLoc();
 
+    Value dest = shuffleOp.getDest();
+
+    // If the destination is a tensor.empty, replace it with an alloc_tensor.
+    if (auto emptyDest = shuffleOp.getDest().getDefiningOp<tensor::EmptyOp>()) {
```

Comment:
Instead of checking if the destination is a `tensor.empty` can we directly use a `bufferization.alloc_tensor` to start with? As I understand it the `tensor.empty` is created when you generate the shuffle tensor op.

Reply:
No, the `tensor.empty` comes from the original LinalgOp destination that ends up as the destination of the `scf.forall`. The destination of the `scf.forall` then becomes the destination of the `shuffle_tensor` op. Rather than further complicating the loop fusion pattern, I'd rather just create the alloc_tensor here.

Reply:
Hmmm... I understand, but It might actually be better to do this during the fusion. IIRC shuffle tensor is introduced when you "fuse" the scf.forall ops. So then it might be better to just introduce the `bufferization.alloc_tensor` there and just drop the `tensor.empty`. It becomes dead code. (But I might be missing something in the overall flow.

Ill unblock for now, but just left comment.

Reply:
Ok, I think the semantics of `shuffle_tensor` are clear enough where allocating at this point is fine, but I see your point about not delaying things. Another option I was considering was just removing the destination from `shuffle_tensor` altogether so I may end up reworking this lowering at a later point anyway.